### PR TITLE
Upgrade to Elixir v1.14

### PR DESCRIPTION
**Pull request by Benjamin-Philip**
This commit upgrades the Elixir version used in the Docker image and the Continuous Integration to v1.14.

### CODE REVIEW

**Comment by doomspork** [comment]
@Benjamin-Philip let's make that change and give it a look to make sure everything works as expected 👍 

Thanks for the PR!

**Comment by Benjamin-Philip** [comment]
@doomspork On investigation I found that the warning:
```
Application.get_env/3 is discouraged in the module body, use Application.compile_env/3 instead
```
is mostly from AppSignal and is passed when we use `AppSignal.Phoenix.view` in `SchoolHouseWeb.view`. It is also passed when we use `AppSignal.Phoenix` in `SchoolHouseWeb.Endpoint`.

Additionally using `Gettext` in SchoolHouseWeb.Gettext passes one instance of the warning.

Since this is something that needs changes upstream, there are 3 things we can do:

1. Close the PR
2. Upgrade to patched versions of these packages (I'll probably have to send the patches myself)
3. Stop using the `--warnings-as-errors` flag temporarily
4. Configure Elixir to ignore this warning (I'm not sure if this is possible though)

This is possibly a bug in the Elixir compiler as these are errors from using an external module.

**Comment by Benjamin-Philip** [comment]
Scratch that. Upgrading `gettext` and `appsignal_phoenix` fixed everything.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ on:
 
 env:
   CACHE_VERSION: v1
-  ELIXIR_VERSION: '1.13'
+  ELIXIR_VERSION: '1.14'
```
Comment (kinson) [inline .github/workflows/ci.yml:16]:
```suggestion
  ELIXIR_VERSION: '1.14.1'
```

File: Dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #   - https://pkgs.org/ - resource for finding needed packages
 #   - Ex: hexpm/elixir:1.13.4-erlang-25.0.4-debian-bullseye-20220801-slim
 #
-ARG ELIXIR_VERSION=1.13.4
+ARG ELIXIR_VERSION=1.14.0
```
Comment (kinson) [inline Dockerfile:15]:
```suggestion
ARG ELIXIR_VERSION=1.14.1
```

**Review by kinson — COMMENTED** [review]
Thanks for digging into the upgrade issues and fixing them 💪🏻 !

Just two quick suggestions:
1. I know this would not have been possible when you opened the pr because it was [only released](https://github.com/elixir-lang/elixir/releases/tag/v1.14.1) 14 days ago, but do you mind updating the version to `1.14.1` so we're on the latest minor version as well?
2. Do you mind updating the `.tool-versions` file so folks that use [asdf](https://github.com/asdf-vm/asdf-elixir) automatically have the correct version set, here is what the file looks like for `1.14.1`:

```shell
elixir 1.14.1 1.14.1-otp-25
```

**Comment by Benjamin-Philip** [comment]
Done.

I haven't tested with asdf though as I don't use it.

**Review by kinson — APPROVED** [review]
This looks good to me! Thanks for the update 📈 !

**Pull request merged**